cmd/git-sync/main.go

```diff
@@ -585,6 +580,207 @@ func addWorktreeAndSwap(ctx context.Context, gitRoot, dest, branch, rev string,
 	return nil
 }
 
+func updateSubmodules(ctx context.Context, worktreePath string, depth int, submoduleMode string) error {
+	log.V(0).Info("updating submodules")
+
+	updateArgs := submoduleUpdateArgs(depth, submoduleMode)
+	submodulesArgs := append([]string{"submodule", "update", "--init"}, updateArgs...)
+
+	if _, err := runCommand(ctx, worktreePath, *flGitCmd, submodulesArgs...); err != nil {
+		return err
+	}
+
+	// Update submodules with remote tracking enabled
+	if *flSubmodulesRemoteTracking != "" {
```

Comment:
please pass this in as an arg - I know it is a bit tedious, but only main() should use raw flags (and yes I know that is not always true already :)

Reply:
Propose the `submoduleOptions type` which encapsulates the options and passing around with this variable instead of the direct access to the `flSubmodules*` which should be only handled in the `main()` as your suggestion.

Proposed patch in this commit https://github.com/kubernetes/git-sync/pull/266/commits/dcf07c19567111761108bc64b8c7c789c3627bdd